SparkPost/2web2ui
# AC-1904 - Upgrades Matchbox to ~4.8.0

[AC-1904](https://sparkpost.atlassian.net/browse/AC-1904)

### What Changed

- Upgrades Matchbox to 4.8.0
- Replaces use of `component` prop with `as` where appropriate
- Updates `ExternalLink` to simplify icon positioning. For Panel actions, the positioning wasn't rendering correctly. Compare the "Learn More" icon position vs. the "Enable SSO" button (the latter is correct):
<img width="308" alt="Screen Shot 2021-02-17 at 12 58 36 PM" src="https://user-images.githubusercontent.com/3613392/108246850-ed0e6300-711f-11eb-9a85-0c52012bd5c7.png">

### How To Test

- Verify the app works as expected
- Verify positioning of elements on `/signals/analytics` and look for padding problems introduced by the Matchbox update
- Verify components that were using the `component` prop continue to work with the `as` prop in its place
- Verify external links and ensure icon positioning is handled appropriately
- Verify instances of the `component` prop were properly replaced with `as` for `Button` and `UnstyledLink` components

### To Do

- [X] Replace use of the deprecated `component` prop with `as`
